src/autopas/selectors/tuningStrategy/BayesianSearch.h

```diff
@@ -163,16 +175,39 @@ bool BayesianSearch::tune() {
 }
 
 FeatureVector BayesianSearch::sampleOptimalFeatureVector(size_t n, AcquisitionFunctionOption af) {
-  // create n lhs samples
-  std::vector<FeatureVector> samples = FeatureVector::lhsSampleFeatures(n, _rng, *_cellSizeFactors, _traversalOptions,
-                                                                        _dataLayoutOptions, _newton3Options);
+  for (size_t i = 0; i < maxAttempts; ++i) {
+    // create n lhs samples
+    std::vector<FeatureVector> samples = FeatureVector::lhsSampleFeatures(n, _rng, *_cellSizeFactors, _traversalOptions,
+                                                                          _dataLayoutOptions, _newton3Options);
+
+    // map container and calculate all acquisition function values
+    std::map<FeatureVector, double> acquisitions;
+    for (auto &sample : samples) {
+      sample.container = _traversalContainerMap[sample.traversal];
+      acquisitions[sample] = _gp.calcAcquisition(af, sample);
+    }
 
-  // sample minimum of acquisition function
-  auto best = _gp.sampleAquisitionMin(af, samples);
+    // sort by acquisition
+    std::sort(samples.begin(), samples.end(), [&acquisitions](const FeatureVector &f1, const FeatureVector &f2) {
+      return acquisitions[f1] < acquisitions[f2];
+    });
+
+    // find first valid configuration
+    auto best = std::find_if(samples.begin(), samples.end(), [this](const FeatureVector &fv) {
+      return _invalidConfigs.find(fv) == _invalidConfigs.end();
+    });
+
+    if (best != samples.end()) {
+      // valid config found!
+      return *best;
+    } else {
+      // No valid configuration. This should rarely happen.
+      AutoPasLog(debug, "BayesianSearch: Tuning could not generate a valid configuration.");
```

Comment:
```suggestion
      AutoPasLog(critical, "Tuning could not generate a valid configuration.");
```
sounds more like a critical (error) instead of debug message. Or is this a recoverable error?
Also the prefix can be dropped as it will be inserted if the user wants it (verbose logging = on).

Reply:
This could happen because there is actually no valid configuration or by chance. To prevent the simulation from crashing in the second case, I allow the function to sample again up to 'maxAttempts = 10' (maybe find a sensible value for this) times. But it if keeps failing, an exception is thrown.